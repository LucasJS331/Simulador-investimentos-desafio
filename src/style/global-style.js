import { createGlobalStyle } from 'styled-components';

export const GlobalStyle = createGlobalStyle`
    *{
        box-sizing: border-box;
        padding: 0;
        margin: 0;
        font-family: 'arial';
    }
    html{
        font-size: 62.5%;
        scroll-behavior: smooth;
    }
    body{
        font-size: 1.6rem;
    }
    input {
      display: block;
      outline: none;
      border: none;
      border-bottom: 1px solid black;
      background-color: transparent;
      padding: 0.5rem;
      width: 100%;
    }

`;
