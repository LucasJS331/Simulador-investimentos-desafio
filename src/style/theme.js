export const theme = {
  colors: {
    primary_color: '#E5E8EB',
    secondary_color: '#FF7D48',
    secondary_color_hover: '#E16736',
  },
  media: {
    default: '(max-width: 830px)',
  },
};
