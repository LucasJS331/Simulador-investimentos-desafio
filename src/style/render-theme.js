import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { FinantialContext } from '../context/finantial-context';
import { theme } from './theme';

export const renderTheme = (children) => {
  return render(
    <FinantialContext>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </FinantialContext>,
  );
};
