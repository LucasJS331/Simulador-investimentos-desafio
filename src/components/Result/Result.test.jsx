import { screen } from '@testing-library/react';
import { Result } from '.';
import { renderTheme } from '../../style/render-theme';

const finantial_data_mocky = {
  valorFinalBruto: 99,
  aliquotaIR: 99,
  valorPagoIR: 99,
  valorFinalLiquido: 99,
  valorTotalInvestido: 99,
  ganhoLiquido: 99,
};
describe('<Result/>', () => {
  it('should render Result Grid', () => {
    const { container } = renderTheme(
      <Result finantial_data={finantial_data_mocky} />,
    );

    expect(container).toMatchSnapshot();
  });
  it('should render loading result', () => {
    renderTheme(
      <Result finantial_data={finantial_data_mocky} isLoading={true} />,
    );
    const img = screen.getByRole('img', { name: 'loading the result' });
    expect(img).toBeInTheDocument();
    expect(img).toHaveAttribute('src', 'loading.gif');
  });
});
