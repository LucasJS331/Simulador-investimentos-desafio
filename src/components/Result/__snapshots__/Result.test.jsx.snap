// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Result/> should render Result Grid 1`] = `
<div>
  <div
    class="sc-gsDKAQ lprlBs"
  >
    <h3
      class="title-result"
    >
      Simulação
    </h3>
  </div>
  <section
    class="sc-dkPtRN dHibC"
    id="form-result"
  >
    <article
      class="sc-bdvvtL cSbLCz"
    >
      <h5>
        Valor final Bruto
      </h5>
      <p
        class=""
        id="article-text"
      >
        99
      </p>
    </article>
    <article
      class="sc-bdvvtL cSbLCz"
    >
      <h5>
        Aliquota do IR
      </h5>
      <p
        class=""
        id="article-text"
      >
        99
      </p>
    </article>
    <article
      class="sc-bdvvtL cSbLCz"
    >
      <h5>
        Valor pago do IR
      </h5>
      <p
        class=""
        id="article-text"
      >
        99
      </p>
    </article>
    <article
      class="sc-bdvvtL cSbLCz"
    >
      <h5>
        Valor final Liquido
      </h5>
      <p
        class="greenValue"
        id="article-text"
      >
        99
      </p>
    </article>
    <article
      class="sc-bdvvtL cSbLCz"
    >
      <h5>
        Valor total Investido
      </h5>
      <p
        class=""
        id="article-text"
      >
        99
      </p>
    </article>
    <article
      class="sc-bdvvtL cSbLCz"
    >
      <h5>
        Ganho liquido
      </h5>
      <p
        class="greenValue"
        id="article-text"
      >
        99
      </p>
    </article>
  </section>
</div>
`;
